### PR TITLE
fix PC-8801/9801 clocks

### LINKED ISSUE #8560: NEC PC-8801 clock
For NEC PC-8801 machines, the master clock is currently defined to exactly 4.0 MHz.

However, according to this blog, is should be 3993600 Hz: http://www7b.biglobe.ne.jp/~crazyunit/pc88.html#clock

I have looked at a few PC-8801 emulators and their clock values:
- M88 emulator: 3993600 Hz
- XM8 emulator: 3993624 Hz

I don't have a definitive source for the actual PC-8801 clock, but exact 4 MHz seems to be a bit off compared to what is documented, so it may be a good idea to fix it to one of the values above.

### PR DESCRIPTION
**Pull request by ValleyBell**
closes #8560, which also served as reference for the clock info

### CODE REVIEW

File: src/mame/drivers/pc88va.cpp [hunk 1]
```diff
@@ -1573,14 +1577,14 @@ void pc88va_state::dma_memw_cb(offs_t offset, uint8_t data)
...
-	V50(config, m_maincpu, 8000000); // μPD9002, aka V30 + μPD70008AC (for PC8801 compatibility mode)
+	V50(config, m_maincpu, MASTER_CLOCK); // μPD9002, aka V30 + μPD70008AC (for PC8801 compatibility mode)
 	m_maincpu->set_addrmap(AS_PROGRAM, &pc88va_state::pc88va_map);
 	m_maincpu->set_addrmap(AS_IO, &pc88va_state::pc88va_io_map);
 	m_maincpu->set_vblank_int("screen", FUNC(pc88va_state::pc88va_vrtc_irq));
 	m_maincpu->set_irq_acknowledge_callback("pic8259_master", FUNC(pic8259_device::inta_cb));
 
 #if TEST_SUBFDC
-	z80_device &fdccpu(Z80(config, "fdccpu", 8000000));        /* 8 MHz */
+	z80_device &fdccpu(Z80(config, "fdccpu", MASTER_CLOCK));        /* 8 MHz */
```
Comment (angelosa) [inline src/mame/drivers/pc88va.cpp:1587]: This is 4 MHz, it’s a PC-80S31K or whatever internal/external variety. 
PS: Arguably the hookup can be converted to use the device itself now but iirc “intelligent mode” is for PC-8801 compatibility SWs anyway.

File: src/mame/drivers/pc9821.cpp [hunk 2]
```diff
@@ -637,14 +637,16 @@ MACHINE_RESET_MEMBER(pc9821_state,pc9821)
 // TODO: setter for DMAC clock should follow up whatever is the CPU clock
 
 // TODO: remove me, cfr. pc9801.cpp
-#define MAIN_CLOCK_X2 2'457'600
+#define BASE_CLOCK      XTAL(31'948'800)    // verified for PC-9801RS/FA, TODO: verified for 9821 series
```
Comment (angelosa) [inline src/mame/drivers/pc9821.cpp:640]: `TODO: verified` sounds like a typo, should be `TODO: verify for PC-9821`?
Also I suggest to merge this TODO with the one just above (IDEs generally expect to read these with a `// TODO: (.*)` pattern.

File: src/mame/drivers/pc88va.cpp (hunk 1 above)
Comment (ValleyBell) [reply src/mame/drivers/pc88va.cpp:1587]: So you mean it *should* be 4 MHz despite what the comment says?

Or did you think that `MASTER_CLOCK` was defined as 4 MHz? (It is defined to `XTAL(8'000'000)` in pc88va.cpp.)

File: src/mame/drivers/pc9821.cpp (hunk 2 above)
Comment (ValleyBell) [reply src/mame/drivers/pc9821.cpp]: Ah, that's indeed a typo - thanks!  
It should be fixed now.

File: src/mame/drivers/pc88va.cpp (hunk 1 above)
Comment (angelosa) [reply src/mame/drivers/pc88va.cpp:1587]: It’s an independent device, running on a separate xtal. PC-88VA is most likely internal, but it was originally an external box.
image

Comment (angelosa) [reply src/mame/drivers/pc88va.cpp:1587]: https://github.com/mamedev/mame/blob/6ffeb38cacc43aca30b7d1468b6f2407635d6d2c/src/mame/machine/pc80s31k.cpp#L40

Comment (ValleyBell) [reply src/mame/drivers/pc88va.cpp]: I reverted the changes related to the floppy device for now.

Going by the discussion from #8560 we shouldn't change it from 8 MHz to 4 MHz unless we have proof that it really runs at 4 MHz or 3.99 MHz.

**Pull request merged**